### PR TITLE
feat: toEsObservable and SymbolShim

### PR DESCRIPTION
**Pull request by gdi2290**

### CODE REVIEW

**Comment by michaelbpaulson** [comment]
@gdi2290 
3 things:
1: there is some lint issues, please fix.
2: could you make the spacing in the symbol file be 4 spaces instead of 2!  Consistency.
3: How do we unit test this?

File: lib/SymbolShim.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+var objectTypes = {
+    "boolean": false,
+    "function": true,
+    "object": true,
+    "number": false,
+    "string": false,
+    "undefined": false
+};
+
+var root = (objectTypes[typeof self] && self) || (objectTypes[typeof window] && window);
```
Comment (gdi2290) [inline lib/SymbolShim.js]: we probably need to tell the linter to ignore these

File: lib/SymbolShim.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+};
+
+/*eslint-disable */
+var _root = (objectTypes[typeof self] && self) || (objectTypes[typeof window] && window);
+var freeGlobal = objectTypes[typeof global] && global;
+
+if (freeGlobal && (freeGlobal.global === freeGlobal || freeGlobal.window === freeGlobal)) {
+    _root = freeGlobal;
+}
+/*eslint-enable */
```
Comment (gdi2290) [inline lib/SymbolShim.js]: I'm suppressing eslint here since we need a hacky way to determine context

**Comment by michaelbpaulson** [comment]
@gdi2290 to address point 3:  Unit tests, they are crashing!

**Comment by michaelbpaulson** [comment]
@jhusain @gdi2290 At this point the unit tests in for properly aborting dataSources is in, you should be able to add the unit tests for rx 5 and get coverage up.  Once that is done, I'll merge this into master.

**Comment by gdi2290** [comment]
awesome, I'll check it out tomorrow

**Comment by michaelbpaulson** [comment]
Going to close this down as it now will be done in part of RxJS.  Then we just require in the module.

**Comment by gdi2290** [comment]
Sounds good

**Pull request closed without merging**